fd/diff.go

```diff
@@ -107,15 +119,29 @@ func Gradient(dst []float64, f func([]float64) float64, x []float64, settings *S
 		panic("fd: slice length mismatch")
 	}
 	if settings == nil {
-		settings = DefaultSettings()
+		settings = &Settings{}
+	}
+	formula := settings.Formula
+	if formula.isZero() {
+		formula = Central
+	}
+	if formula.Order == 0 || formula.Stencil == nil || formula.Step == 0 {
+		panic("fd: bad formula")
 	}
-	step := settings.Method.Step
+	if formula.Order != 1 {
+		panic("fd: invalid derivative order")
+	}
+	step := settings.Step
+	if step == 0 {
+		step = formula.Step
+	}
+
```

Comment:
Maybe move the nWorkers code up here, and then if nWorkers == 1 do the serial code